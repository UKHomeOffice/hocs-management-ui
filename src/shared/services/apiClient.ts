import axios from 'axios';

export default {
    createClient: () => {
        const client = axios.create();

        return {
            get: client.get,
            delete: client.delete,
            post: client.post,
            put: client.put
        };
    }
};
