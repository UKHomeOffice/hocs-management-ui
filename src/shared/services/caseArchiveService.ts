import axios from 'axios';
import ArchiveCaseModel from '../models/archiveCaseModel';

export const archiveCase = (archiveCaseModel: ArchiveCaseModel) => new Promise((resolve, reject) => axios
    .post('/api/case/archive', archiveCaseModel)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
