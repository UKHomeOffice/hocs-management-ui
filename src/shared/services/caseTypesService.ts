import axios from 'axios';
import CaseType from '../models/caseType';
import Item from '../models/item';

const sortByLabel: ((a: Item, b: Item) => number) = (a, b) => {
    const labelA = a.label.toLocaleLowerCase();
    const labelB = b.label.toLocaleLowerCase();
    let comparison = 0;
    if (labelA > labelB) {
        comparison = 1;
    } else if (labelA < labelB) {
        comparison = -1;
    }
    return comparison;
};

export const getCaseTypes = () => new Promise<CaseType[]>((resolve, reject) =>
    axios.get('/api/case-types')
        .then((response) => {
            resolve(response.data.sort(sortByLabel));
        })
        .catch(error => reject(error))
);
