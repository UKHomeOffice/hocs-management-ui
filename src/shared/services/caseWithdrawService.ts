import axios from 'axios';
import WithdrawCaseModel from '../models/withdrawCaseModel';

export const withdrawCase = (withdrawCaseModel: WithdrawCaseModel) => new Promise((resolve, reject) => axios
    .post('/api/case/withdraw', withdrawCaseModel)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
