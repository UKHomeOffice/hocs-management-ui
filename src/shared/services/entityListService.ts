import axios from 'axios';
import EntityListItem from 'shared/models/entityListItem';

export const getListItems = (listName: string) => new Promise<EntityListItem[]>((resolve, reject) => axios
    .get(`/api/entity/list/${listName}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const createListItem = (item: EntityListItem, listName: string) => new Promise((resolve, reject) => axios
    .post(`/api/entity/list/${listName}`, item)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);

export const getItemDetails = (itemUUID: string) => new Promise<EntityListItem>((resolve, reject) => axios
    .get(`/api/entity/${itemUUID}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const updateListItem = (item: EntityListItem, listName: string) => new Promise((resolve, reject) => axios
    .post(`/api/entity/list/update/${listName}`, item)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);

export const deleteListItem = (uuid: string, listName: string) => new Promise((resolve, reject) => axios
    .delete(`/api/entity/list/${listName}`, { data: { entityUUID: uuid } })
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
