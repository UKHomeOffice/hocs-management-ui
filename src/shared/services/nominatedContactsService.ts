import axios from 'axios';
import NominatedContact from "../models/nominatedContact";

export const addNominatedContact = (nominatedContact: NominatedContact) => new Promise((resolve, reject) => axios
    .post('/api/nominated-contact', nominatedContact)
    .then(() => resolve())
    .catch(reason => reject(reason))
);
