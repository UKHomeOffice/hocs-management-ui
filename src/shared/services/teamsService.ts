import axios from 'axios';
import Item from '../models/item';
import Team from '../models/team';

export const getTeam = (teamId: string) => new Promise<Team>((resolve, reject) =>
    axios.get(`/api/teams/${teamId}`)
        .then(response => resolve(response.data))
        .catch(error => reject(error))
);

export const getTeams = () => new Promise<[Item]>((resolve, reject) =>
    axios.get('/api/teams')
        .then(response => resolve(response.data))
        .catch(error => reject(error))
);

export const getTeamMembers = (teamId: string) => new Promise((resolve, reject) => axios
    .get(`/api/teams/${teamId}/members`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const getTeamsForUser = (userId: string) => new Promise<[Item]>((resolve, reject) => axios
    .get(`/api/teams/${userId}/teams`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const createTeam = (team: Team) => new Promise((resolve, reject) => axios
    .post(`/api/teams/unit/${team.unitUUID}`, team)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);
