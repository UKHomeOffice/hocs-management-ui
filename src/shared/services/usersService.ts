import apiClient from './apiClient';
import Item from '../models/item';

const usersApi = apiClient.createClient();

export class AddUserError extends Error {
    userToAdd: Item;
    constructor(message: string, userToAdd: Item) {
        super(message);
        this.userToAdd = userToAdd;
    }
}

export const getUsers = () => usersApi.get('/api/users');

export const addUserToTeam = (user: Item, teamId: string) => {
    return new Promise((resolve, reject) => usersApi
        .post(`/api/users/${user}/team/${teamId}`)
        .then(value => resolve(value.data))
        .catch(reason => reject(new AddUserError(reason, user))));
};

export const deleteUserFromTeam = (userId: string, teamId: string) => usersApi.delete(`/api/users/${userId}/team/${teamId}`);
