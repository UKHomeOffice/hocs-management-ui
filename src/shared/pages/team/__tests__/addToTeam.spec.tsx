import React from 'react';
import { MemoryRouter, match } from 'react-router';
import { createBrowserHistory, History, Location } from 'history';
import { act, render, RenderResult } from '@testing-library/react';
import AddToTeam from '../addToTeam';
import * as TeamsService from '../../../services/teamsService';
import * as UsersService from '../../../services/usersService';

let match: match<any>;
let history: History<any>;
let location: Location;

const getTeamSpy = jest.spyOn<typeof TeamsService, 'getTeam'>(TeamsService, 'getTeam').mockReturnValue(Promise.resolve({
    active: true,
    displayName: '__displayName__',
    letterName: '__letterName__',
    permissions: [],
    type: '__type__'
}));

const getUsersSpy = jest.spyOn(UsersService, 'getUsers')
    .mockReturnValue(Promise.resolve({
        data: [{
            label: '__user1__',
            value: '__userId1__'
        }, {
            label: '__user2__',
            value: '__userId2__'
        }]
    }));

beforeEach(() => {
    history = createBrowserHistory();
    match = {
        isExact: true,
        params: { teamId: '__teamId__' },
        path: '',
        url: ''
    };

    location = {
        hash: '',
        key: '',
        pathname: '',
        search: '',
        state: {}
    };
});

describe('addToTeam component', () => {
    it('should render with default props', async () => {
        let wrapper: RenderResult;
        act(() => {
            wrapper = render(<MemoryRouter><AddToTeam history={history} location={location} match={match}></AddToTeam></MemoryRouter>);
        });
        // @ts-ignore
        return getTeamSpy('__teamId__').then(() => {
            expect(getTeamSpy).toHaveBeenCalled();
            expect(getUsersSpy).toHaveBeenCalled();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
