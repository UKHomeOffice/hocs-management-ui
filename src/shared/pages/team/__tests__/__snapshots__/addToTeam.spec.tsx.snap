// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addToTeam component should render with default props 1`] = `""`;
